crates/re_log_types/src/serde_field.rs

```diff
@@ -0,0 +1,126 @@
+use arrow2::{
+    array::{BinaryArray, MutableBinaryArray, TryPush},
+    datatypes::DataType,
+};
+use arrow2_convert::{deserialize::ArrowDeserialize, field::ArrowField, serialize::ArrowSerialize};
+
+/// Helper for storing arbitrary serde-compatible types in an arrow2 [`BinaryArray`] field
+///
+/// Use as:
+/// ```
+/// use re_log_types::serde_field::SerdeField;
+/// use arrow2_convert::{ArrowDeserialize, ArrowField, ArrowSerialize, field::ArrowField};
+/// use arrow2::datatypes::{DataType, Field};
+///
+/// #[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+/// struct SomeStruct {
+///     foo: String,
+///     bar: u32,
+/// }
+///
+/// #[derive(Clone, Debug, PartialEq, ArrowField, ArrowSerialize, ArrowDeserialize)]
+/// struct SomeArrowStruct {
+///     #[arrow_field(type = "SerdeField<SomeStruct>")]
+///     field: SomeStruct,
+/// }
+///
+/// assert_eq!(
+///     SomeArrowStruct::data_type(),
+///     DataType::Struct(vec![
+///         Field::new("field", DataType::Binary, false),
+///     ])
+/// );
+/// ```
+pub struct SerdeField<T>(std::marker::PhantomData<T>);
```

Comment:
I wonder how we should handle the fact that we use MsgPack as the "codec". There are alternatives, like `bincode`, we could be using, and might want to switch too in the future.

We could, for instance, add an `enum Codec { MsgPack = 1 }` and put that as the first byte of the encoded array.

Reply:
I think the "right" thing to do in theory is to add this to the arrow meta-data for the field. We've been wanting to use that for some other stuff as well so this might be a good opportunity to investigate the plumbing.